packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/query/ScalarQuery.kt

```diff
@@ -140,31 +142,35 @@ internal class MinMaxQuery<E : BaseRealmObject, T : Any> constructor(
     }
 
     @Suppress("ComplexMethod")
-    private fun computeAggregatedValue(resultsPointer: RealmResultsPointer, propertyKey: PropertyKey): T? {
-        val result: T? = when (queryType) {
+    private fun computeAggregatedValue(
+        resultsPointer: RealmResultsPointer,
+        propertyKey: PropertyKey
+    ): T? = getterScope {
+        val transport = when (queryType) {
             AggregatorQueryType.MIN ->
-                RealmInterop.realm_results_min(resultsPointer, propertyKey).value as T?
+                RealmInterop.realm_results_min(allocRealmValueT(), resultsPointer, propertyKey)
             AggregatorQueryType.MAX ->
-                RealmInterop.realm_results_max(resultsPointer, propertyKey).value as T?
+                RealmInterop.realm_results_max(allocRealmValueT(), resultsPointer, propertyKey)
             AggregatorQueryType.SUM ->
                 throw IllegalArgumentException("Use SumQuery instead.")
         }
-        // TODO Expand to support other numeric types, e.g. Decimal128
+
         @Suppress("UNCHECKED_CAST")
-        return when (result) {
+        return when (transport) {
             null -> null
-            is Timestamp -> RealmInstant.from(result.seconds, result.nanoSeconds)
-            is Number -> when (type) {
-                Int::class -> result.toInt()
-                Short::class -> result.toShort()
-                Long::class -> result.toLong()
-                Float::class -> result.toFloat()
-                Double::class -> result.toDouble()
-                Byte::class -> result.toByte()
-                Char::class -> result.toChar()
-                else -> throw IllegalArgumentException("Invalid numeric type for '$property', it is not a '${type.simpleName}' or cannot be represented by it.")
+            else -> when (type) {
+                Int::class -> transport.getLong().toInt()
+                Short::class -> transport.getLong().toShort()
+                Long::class -> transport.getLong()
+                Float::class -> transport.getFloat()
+                Double::class -> transport.getDouble()
+                Byte::class -> transport.getLong().toByte()
+                Char::class -> transport.getLong().toInt().toChar()
+                RealmInstant::class -> transport.getTimestamp().let { timestamp ->
+                    RealmInstant.from(timestamp.seconds, timestamp.nanoSeconds)
+                }
```

Comment:
Yeah, I said that in the gist I shared before 😅 . We can instantiate the datatypes directly from the SDK because we have access to the realm_value_t.

But I agree it is more pressing to close the PR and move forward, we can address these things later on.